src/compiler/transformers/transform-utils.ts

```diff
@@ -531,7 +531,38 @@ const getTypeReferenceLocation = (typeName: string, tsNode: ts.Node): d.Componen
   };
 };
 
-export const resolveType = (checker: ts.TypeChecker, type: ts.Type) => {
+/**
+ * Resolve a type annotation, using the TypeScript typechecker to convert a
+ * {@link ts.Type} record to a string.
+ *
+ * For instance, if a module `bar.ts` imports a type `Foo` from another file
+ * which looks like this:
+ *
+ * ```ts
+ * // foo.ts
+ * type Foo = (b: bar) => boolean;
```

Comment:
Can we do one of two things? Either update `bar` to be capital cased (`Bar`) here and in the third code block, or could be change this to a primitive value (like `number`)? IDK if having `Bar` here necessarily helps communicate how the type annotation gets resolved (as I'm not sure where `bar` originates from in this example, since it's first introduced in `foo.ts` and not defined in either `foo.ts` or `bar.ts`)

Reply:
yeah I think it should be just `string` or something, I think I had a more complicated example earlier which I didn't totally excise

Reply:
ended up rewording / reworking that a little bit to hopefully make it more clear, lmk what you think